<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsDialog</class>
 <widget class="QDialog" name="SettingsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>750</width>
    <height>550</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_main">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QListWidget" name="settingsListWidget">
     <property name="maximumSize">
      <size>
       <width>180</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="currentRow">
      <number>0</number>
     </property>
     <item>
      <property name="text">
       <string>General</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Translation</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>AI Engine</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Plugins</string>
      </property>
     </item>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_content">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_content">
      <property name="spacing">
       <number>12</number>
      </property>
      <property name="leftMargin">
       <number>12</number>
      </property>
      <property name="topMargin">
       <number>12</number>
      </property>
      <property name="rightMargin">
       <number>12</number>
      </property>
      <property name="bottomMargin">
       <number>12</number>
      </property>
      <item>
       <widget class="QStackedWidget" name="configStackedWidget">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <!-- Page 0: General (Source/Target, Mode, AI Toggle) -->
        <widget class="QWidget" name="page_general">
         <layout class="QVBoxLayout" name="verticalLayout_general">
          <item>
           <widget class="QGroupBox" name="generalGroupBox">
            <property name="title">
             <string>General Settings</string>
            </property>
            <layout class="QFormLayout" name="formLayout_general">
             <property name="horizontalSpacing">
              <number>15</number>
             </property>
             <property name="verticalSpacing">
              <number>10</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="sourceLanguageLabel">
               <property name="text">
                <string>Source Language:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QComboBox" name="sourceLanguageComboBox"/>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="targetLanguageLabel">
               <property name="text">
                <string>Target Language:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QComboBox" name="targetLanguageComboBox"/>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="activeModeLabel">
               <property name="text">
                <string>Active Mode:</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QComboBox" name="translatorModeComboBox">
               <item>
                <property name="text">
                 <string>Quick (Google Free)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Professional (Google API)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>AI-Powered (LLM)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Plugins (Lua)</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label_enable_ai">
               <property name="text">
                <string>AI Features:</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QCheckBox" name="enableAiFilterCheckBox">
               <property name="text">
                <string>Enable AI Context Filter</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_general">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        
        <!-- Page 1: Translation (Engine Configs) -->
        <widget class="QWidget" name="page_translation">
         <layout class="QVBoxLayout" name="verticalLayout_translation">
          <item>
           <widget class="QScrollArea" name="scrollArea_translation">
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <widget class="QWidget" name="scrollAreaWidgetContents_translation">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>100</width>
               <height>100</height>
              </rect>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_scroll_translation">
              <item>
               <widget class="QGroupBox" name="googleProGroupBox">
                <property name="title">
                 <string>Google Cloud Translation API</string>
                </property>
                <layout class="QFormLayout" name="formLayout_professional">
                 <item row="0" column="0">
                  <widget class="QLabel" name="googleApiKeyLabel">
                   <property name="text">
                    <string>API Key:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QLineEdit" name="googleApiKeyEdit"/>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="llmProviderGroup">
                <property name="title">
                 <string>LLM Provider Settings</string>
                </property>
                <layout class="QFormLayout" name="formLayout_ai_powered">
                 <item row="0" column="0">
                  <widget class="QLabel" name="llmProviderLabel">
                   <property name="text">
                    <string>Provider:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QComboBox" name="llmProviderComboBox">
                   <item>
                    <property name="text">
                     <string>Google AI</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>Anthropic</string>
                    </property>
                   </item>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="llmApiKeyLabel">
                   <property name="text">
                    <string>API Key:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QLineEdit" name="llmApiKeyEdit"/>
                 </item>
                 <item row="2" column="0">
                  <widget class="QLabel" name="llmModelLabel">
                   <property name="text">
                    <string>Model:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="QComboBox" name="llmModelComboBox"/>
                 </item>
                 <item row="3" column="0" colspan="2">
                  <widget class="QGroupBox" name="llmAdvancedGroupBox">
                   <property name="title">
                    <string>Advanced</string>
                   </property>
                   <property name="checkable">
                    <bool>true</bool>
                   </property>
                   <property name="checked">
                    <bool>false</bool>
                   </property>
                   <layout class="QFormLayout" name="formLayout_advanced">
                    <item row="0" column="0">
                     <widget class="QLabel" name="llmBaseUrlLabel">
                      <property name="text">
                       <string>Base URL:</string>
                      </property>
                     </widget>
                    </item>
                    <item row="0" column="1">
                     <widget class="QLineEdit" name="llmBaseUrlEdit"/>
                    </item>
                   </layout>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer_translation">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>

        <!-- Page 2: AI (UI) & Visuals -->
        <widget class="QWidget" name="page_ai_ui">
         <layout class="QVBoxLayout" name="verticalLayout_ai_ui">
          <item>
           <widget class="QGroupBox" name="aiFilterDetailsGroupBox">
            <property name="title">
             <string>AI Filter Configuration</string>
            </property>
            <layout class="QFormLayout" name="formLayout_ai_settings">
             <item row="0" column="0">
              <widget class="QLabel" name="label_ai_sensitivity">
               <property name="text">
                <string>Sensitivity:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QDoubleSpinBox" name="aiFilterThresholdSpinBox">
               <property name="maximum">
                <double>1.0</double>
               </property>
               <property name="singleStep">
                <double>0.05</double>
               </property>
               <property name="value">
                <double>0.75</double>
               </property>
               <property name="toolTip">
                <string>Similarity Threshold (Higher = Stricter/Fewer Skips)</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0" colspan="2">
              <widget class="QLabel" name="aiInfoLabel">
               <property name="text">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;Details on how the AI interprets the image context.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="wordWrap">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="visualsGroupBox">
            <property name="title">
             <string>Visualizations</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_visuals">
             <item>
              <widget class="QCheckBox" name="enableRelationsCheckBox">
               <property name="text">
                <string>Enable Data Relations Graph (RPG Maker)</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_ai_ui">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>

        <!-- Page 3: Plugins -->
        <widget class="QWidget" name="page_plugins">
         <layout class="QHBoxLayout" name="horizontalLayout_plugins">
          <item>
           <widget class="QListWidget" name="pluginListWidget">
            <property name="maximumSize">
             <size>
              <width>150</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_plugins_right">
            <item>
             <widget class="QCheckBox" name="pluginEnabledCheckBox">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Enable Plugin</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QScrollArea" name="pluginSettingsScrollArea">
              <property name="widgetResizable">
               <bool>true</bool>
              </property>
              <widget class="QWidget" name="pluginSettingsContainer">
               <property name="geometry">
                <rect>
                 <x>0</x>
                 <y>0</y>
                 <width>256</width>
                 <height>228</height>
                </rect>
               </property>
               <layout class="QFormLayout" name="formLayout_plugins"/>
              </widget>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QDialogButtonBox" name="buttonBox">
        <property name="standardButtons">
         <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>accept()</slot>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>reject()</slot>
  </connection>
  <connection>
   <sender>settingsListWidget</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>configStackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
  </connection>
 </connections>
</ui>